swimster-ui/src/components/CreatePoolSteps/CreatePoolSteps.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react'
+import { amenities } from '../Amenities/Amenities'
+
+const style = {
+    formCreatePoolHeader: 'font-semibold text-3xl col-span-3 mb-3',
+    formAmenitiesHeader: 'font-semibold text-2xl col-span-3',
+    formDimensionsHeader: 'font-semibold text-2xl col-span-3',
+    amenitiesContainer: 'col-span-3',
+    amenitiesContent: 'grid grid-cols-3 gap-6 mt-3',
+    amenityContainer: 'rounded-xl flex flex-col space-y-2 items-center border px-3 py-4 font-semibold cursor-pointer',
+    selectedAmenity: 'bg-gray-500 text-gray-200 border-gray-500',
+    checked: 'bg-blue-100',
+    amenityTitle: 'text-sm',
+    amenityImage: 'w-8 h-8 fill-gray-200',
+    inputElement: 'rounded-md p-3 border-none outline-none bg-gray-200 ring-gray-400 hover:ring-2 focus:ring-4 focus:ring-offset-2 focus:ring-offset-gray-300',
+    fullInput: 'col-span-3 resize-none',
+    formStepContinueButton: 'bg-blue-700 py-3 col-start-3 rounded-md mt-4 cursor-pointer font-semibold text-gray-200',
+    formStepBackButton: 'bg-blue-700 py-3 col-start-2 col-span-1 rounded-md mt-4 cursor-pointer font-semibold text-gray-200',
+    formStepConfirmButton: 'bg-blue-700 py-3 col-start-3 col-span-1 rounded-md mt-4 cursor-pointer font-semibold text-gray-200',
+}
+
+const CreatePoolSteps = ({ step, nextStep, prevStep, register, setValue, resetField }) => {
+    const [amenitiesChosen, setAmenitiesChosen] = useState([])
+
+    // allows selecting and deselecting of amenities
+    const handleSelected = (amenity) => {
+        if (getSelected(amenity)) {
+            const newSeletedAmenities = amenitiesChosen.filter(amenityTitle => amenityTitle !== amenity.title)
+            return setAmenitiesChosen(newSeletedAmenities)
+        }
+        setAmenitiesChosen([...amenitiesChosen, amenity.title])
+    }
+
+    // checks if amenity is selected for UI change
+    const getSelected = amenity => amenitiesChosen.includes(amenity.title)
+
+    // resets these fields - solves autocomplete fields bug
+    useEffect(() => {
+        resetField("poolLength")
+        resetField("poolWidth")
+        resetField("poolDepth")
```

Comment:
amenities don't need reset?

Reply:
I can reset it just for consistency, but I only did that with these fields because they were causing a bug where those fields would auto populate with the values from the last step in the form

Reply:
I tried resetting it but since it's not a natural `input` element, it doesn't do anything anyway. I also like that it saves your selections if you have to go back and change something. Again, I only did this because of the bug explained above.